Modules/_testexternalinspection.c

```diff
@@ -1464,6 +1471,240 @@ find_running_task(
     return 0;
 }
 
+static int
+append_awaited_by_for_thread(
+    int pid,
+    uintptr_t head_addr,
+    struct _Py_DebugOffsets *debug_offsets,
+    struct _Py_AsyncioModuleDebugOffsets *async_offsets,
+    PyObject *result
+) {
+    struct llist_node task_node;
+
+    if (0 > read_memory(
+                pid,
+                head_addr,
+                sizeof(task_node),
+                &task_node))
+    {
+        return -1;
+    }
+
+    while ((uintptr_t)task_node.next != head_addr) {
+        uintptr_t task_addr = (uintptr_t)task_node.next
+            - async_offsets->asyncio_task_object.task_node;
+
+        PyObject *tn = parse_task_name(
+            pid,
+            debug_offsets,
+            async_offsets,
+            task_addr);
+        if (tn == NULL) {
+            return -1;
+        }
+
+        PyObject *current_awaited_by = PyList_New(0);
+        if (current_awaited_by == NULL) {
+            Py_DECREF(tn);
+            return -1;
+        }
+
+        PyObject *result_item = PyTuple_New(2);
+        if (result_item == NULL) {
+            Py_DECREF(tn);
+            Py_DECREF(current_awaited_by);
+            return -1;
+        }
+
+        PyTuple_SET_ITEM(result_item, 0, tn);  // steals ref
+        PyTuple_SET_ITEM(result_item, 1, current_awaited_by);  // steals ref
+        if (PyList_Append(result, result_item)) {
+            Py_DECREF(result_item);
+            return -1;
+        }
+        Py_DECREF(result_item);
```

Comment:
A bit slower because parsing the format is worse but we can consider doing

```suggestion
PyObject *result_item = Py_BuildValue("(NN)", tn, current_awaited_by);
if (result_item == NULL) {
    Py_DECREF(tn);
    Py_DECREF(current_awaited_by);
    return -1;
}
```

Reply:
I like that your version is simpler, but I'd prefer not to do something slower since we're already fighting for time when remotely accessing memory.